theano/tensor/nnet/corr.py

```diff
@@ -357,15 +367,16 @@ def c_code_helper(self, bottom, weights, top, sub, height=None, width=None):
     }
 
     // Infer output shape
-    npy_intp out_dim[4];
+    npy_intp out_dim[6];
+    out_dim[4] = out_dim[5] = 0; //Will be changed if required
```

Comment:
This is a bit vague. I'd prefer something like: "only used for unshared backprop wrt. weights".